src/braket/circuits/noise.py

```diff
@@ -261,38 +297,27 @@ def __init__(
                 f"{identity} is not allowed as a key. Please enter only non-identity Pauli strings."
             )
 
+        total_prob = 0
         for pauli_string, prob in probabilities.items():
-            if not isinstance(pauli_string, str):
-                raise TypeError(f"Type of {pauli_string} was not a string.")
-            if len(pauli_string) != self.qubit_count:
-                raise ValueError(
-                    (
-                        "Length of each Pauli string must be equal to number of qubits. "
-                        f"{pauli_string} had length {len(pauli_string)} instead of length {self.qubit_count}."  # noqa
+            MultiQubitPauliNoise._validate_pauli_string(
+                pauli_string, self.qubit_count, self._allowed_substrings
+            )
+            if isinstance(prob, float):
```

Comment:
Should the logic of the conditional statement be similar to the checks on L124? Seems that we are doing similar checks but the logic is arranged slightly differently for each class.